pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -2,138 +2,252 @@
+		} else {
+			return normalizedType, fmt.Sprintf(`\\%s\%s`, m.Host, m.Share)
+		}
+	}
+}
+
+type diskConfig struct {
+	useMount            bool
+	includedDevices     []regexp.Regexp
+	excludedDevices     []regexp.Regexp
+	includedFilesystems []regexp.Regexp
+	excludedFilesystems []regexp.Regexp
+	includedMountpoints []regexp.Regexp
+	excludedMountpoints []regexp.Regexp
+	tagByFilesystem     bool
+	allPartitions       bool
+	deviceTagRe         map[*regexp.Regexp][]string
+	includeAllDevices   bool
+	minDiskSize         uint64
+	tagByLabel          bool
+	useLsblk            bool
+	blkidCacheFile      string
+	serviceCheckRw      bool
+}
 
-		// device name matches `excluded_disk_re`
-		if c.cfg.excludedDiskRe != nil && c.cfg.excludedDiskRe.MatchString(device) {
+func sliceMatchesExpression(slice []regexp.Regexp, expression string) bool {
+	for _, regexp := range slice {
+		if regexp.MatchString(expression) {
 			return true
 		}
 	}
+	return false
+}
 
-	// fs is listed in `excluded_filesystems`
-	if stringSliceContain(c.cfg.excludedFilesystems, fstype) {
-		return true
+type Check struct {
+	core.CheckBase
+	cfg          *diskConfig
+	deviceLabels map[string]string
+}
+
+func newDiskConfig() *diskConfig {
+	return &diskConfig{
+		useMount:            false,
+		includedDevices:     []regexp.Regexp{},
+		excludedDevices:     []regexp.Regexp{},
+		includedFilesystems: []regexp.Regexp{},
+		excludedFilesystems: []regexp.Regexp{},
+		includedMountpoints: []regexp.Regexp{},
+		excludedMountpoints: []regexp.Regexp{},
+		tagByFilesystem:     false,
+		allPartitions:       false,
+		deviceTagRe:         make(map[*regexp.Regexp][]string),
+		includeAllDevices:   true,
+		minDiskSize:         0,
+		tagByLabel:          true,
+		useLsblk:            false,
+		blkidCacheFile:      "",
+		serviceCheckRw:      false,
 	}
+}
 
-	// device mountpoint matches `excluded_mountpoint_re`
-	if c.cfg.excludedMountpointRe != nil && c.cfg.excludedMountpointRe.MatchString(mountpoint) {
-		return true
+func (c *Check) Run() error {
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
 	}
+	if c.cfg.tagByLabel {
+		err = c.fetchAllDeviceLabels()
+		if err != nil {
+			log.Debugf("Unable to fetch device labels: %s", err)
+		}
+	}
+	err = c.collectPartitionMetrics(sender)
+	if err != nil {
+		return err
+	}
+	err = c.collectDiskMetrics(sender)
+	if err != nil {
+		return err
+	}
+	sender.Commit()
 
-	// all good, don't exclude the disk
-	return false
+	return nil
 }
 
-func (c *Check) instanceConfigure(data integration.Data) error {
-	conf := make(map[interface{}]interface{})
-	c.cfg = &diskConfig{}
-	err := yaml.Unmarshal([]byte(data), &conf)
+func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
+	err := c.CommonConfigure(senderManager, initConfig, data, source)
 	if err != nil {
 		return err
 	}
+	return c.diskConfigure(data, initConfig)
```

Comment:
Nitpick: Should rename the private function to `configureDiskCheck` 
It reads less abstract or even `initilizeCheck`